fix build issues with GCC 4.9
based on http://debian-med.debian.narkive.com/gmayw4l4/help-needed-in-c-seqan-issue#post13
autor: Kenneth Hoste (Ghent University)
--- tophat-2.0.12/src/SeqAn-1.3/seqan/file/file_format.h.orig	2015-06-04 21:40:11.418996750 +0200
+++ tophat-2.0.12/src/SeqAn-1.3/seqan/file/file_format.h	2015-06-04 21:40:29.939970826 +0200
@@ -150,27 +150,31 @@
 	~FileFormat() {}
 	FileFormat const & operator =(FileFormat const &) { return *this; }
 
+    // flesh out the following methods in order to
+    // let g++ accept this forward:
+    // seqan/file/file_generated_forwards.h:962:66: error: invalid abstract return type 'seqan::FileFormat<TFile, TData, TMeta, void>'
+    // template <typename TFile, typename TData, typename TMeta> inline FileFormat<TFile, TData, TMeta, void> guessFileFormat(TFile & file, TData & data); // "projects/library/seqan/file/file_form */
 	virtual void *
-	formatID_() const = 0;
+    formatID_() const { return 0; }
 
 	virtual void
-	read_(TFile & file, TData & data) const = 0;
+    read_(TFile & file, TData & data) const {}
 	virtual void
-	read_(TFile & file, TData & data, TSize limit) const = 0;
+    read_(TFile & file, TData & data, TSize limit) const {}
 
 	virtual void
-	readMeta_(TFile & file, TMeta & meta) const = 0;
+    readMeta_(TFile & file, TMeta & meta) const {}
 
 	virtual void
-	goNext_(TFile & file) const = 0;
+    goNext_(TFile & file) const {}
 
 	virtual TSize
-	length_(TFile & file) const = 0;
+    length_(TFile & file) const { return (TSize)-1; }
 
 	virtual void
-	write_(TFile & file, TData & data) const = 0;
+    write_(TFile & file, TData & data) const {}
 	virtual void
-	write_(TFile & file, TData & data, TMeta & meta) const = 0;
+    write_(TFile & file, TData & data, TMeta & meta) const {}
 
 };
 
